lib/grape/exceptions/base.rb

```diff
@@ -71,6 +71,7 @@ def translate_message(key, **options)
       end
 
       def translate(key, **options)
+        options[:default] &&= options[:default].to_s
```

Comment:
This modifies the caller's `options`, which can easily be a frozen hash or whatever.

Reply:
`*` and `**` in arguments act a different way with common argument.

```ruby
def common_argument_test(args)
  p args.object_id
end

def keyword_argument_test(**args)
  p args.object_id
end

args = { a: 1 }
p args.object_id # => 70338004137000
common_argument_test args # => 70338004137000
keyword_argument_test args # => 70338004136780
```
In there code, `args` in `keyword_argument_test ` is a new hash, create by the method itself.I think we can treat it as a most basic hash.

Reply:
I guess we always declared it the old fashioned way as `options = {}`. I worry that this is different version from version, especially reading [this SO](https://stackoverflow.com/questions/18289152/what-does-a-double-splat-operator-do). Can we just do this the old fashioned way and avoid surprises?

Reply:
Do you mean refactoring out the code, for there here are many `**` now?

Reply:
Sorry if I wasn't clear. The double splat is fine. I mean this:

```ruby
        options = options.dup
        options[:default] &&= options[:default].to_s
```

We dup arguments everywhere as far as I can see.

Reply:
It's a really good habit in case of unknown changes. <br>I have added it :)